### PR TITLE
Remove some special cases from dmd.typesem.resolveHelper

### PR DESCRIPTION
**Pull request by BorisCarvajal**

### CODE REVIEW

**Comment by SixthDot** [comment]
you may add #14792 test case if that works, then I'll close it.

**Comment by BorisCarvajal** [comment]
> That seems to work, just the expected output messages need to be updated.

Yes.

There were a bunch of conditions in the code but almost always `helper3()` was called, just noise.
I think it's better this way because we avoid duplicate logic and error messages. If a qualified type is not found, it's tried as an expression (exp semantics are just more powerful).

**Review by ghost — APPROVED** [review]

File: compiler/src/dmd/typesem.d [hunk 1]
```diff
@@ -285,54 +285,10 @@ private void resolveHelper(TypeQualified mt, const ref Loc loc, Scope* sc, Dsymb
...
-                    if (sm)
-                        error(loc, "identifier `%s` of `%s` is not defined, did you mean %s `%s`?", id.toChars(), mt.toChars(), sm.kind(), sm.toChars());
-                    else
-                        error(loc, "identifier `%s` of `%s` is not defined", id.toChars(), mt.toChars());
-                }
-                pe = ErrorExp.get();
-                return;
-            }
-        }
+            return helper3();
```
Comment (ibuclaw) [inline compiler/src/dmd/typesem.d:290]: `helper3` :anguished:

Comment (BorisCarvajal) [reply compiler/src/dmd/typesem.d:290]: Probably it should be renamed to something like `resolveAsExpression`.

**Review by RazvanN7 — APPROVED** [review]

**Pull request merged**